src/tests/JIT/HardwareIntrinsics/General/Shared/VectorDotTest.template

```diff
@@ -324,5 +311,30 @@ namespace JIT.HardwareIntrinsics.General
                 Succeeded = false;
             }
         }
+
+        private {RetBaseType} ComputeExpectedResult(ReadOnlySpan<{Op1BaseType}> left, ReadOnlySpan<{Op2BaseType}> right)
+        {
+            // We need to do pairwise addition to compute the result. However, we need this done for the vector chunks
+            // as well. So we'll just recurse down to 1 element. This ensures that:
+            // * Vector512<T> is Vector256<T> + Vector256<T>
+            // * Vector256<T> is Vector128<T> + Vector128<T>
+            // * Vector128<T> is Vector64<T>  + Vector64<T>
+            // * etc
+            // This continues down to `(left[n] * right[n]) + (left[n + 1] * right[n + 1])`
+
+            if (left.Length != 1)
+            {
+                int half = left.Length / 2;
+
+                return ({RetBaseType})(
+                    ComputeExpectedResult(left.Slice(0, half), right.Slice(0, half)) +
+                    ComputeExpectedResult(left.Slice(half), right.Slice(half))
+                );
+            }
+            else
+            {
+                return ({RetBaseType})(left[0] * right[0]);
+            }
+        }
```

Comment:
Left and Right are always going to have the same number of elements, correct?

Reply:
Yes